src/SA/ldapsearch/entry.c

```diff
@@ -0,0 +1,398 @@
+#include <rpc.h>
+#include <rpcdce.h>
+#include <stdint.h>
+#include "bofdefs.h"
+#include "base.c"
+#include <string.h>
+
+DECLSPEC_IMPORT DWORD WINAPI NETAPI32$DsGetDcNameA(LPVOID, LPVOID, LPVOID, LPVOID, ULONG, LPVOID);
+DECLSPEC_IMPORT DWORD WINAPI NETAPI32$NetApiBufferFree(LPVOID);
+
+DECLSPEC_IMPORT LDAP* WINAPI WLDAP32$ldap_init(PSTR, ULONG);
+DECLSPEC_IMPORT ULONG WINAPI WLDAP32$ldap_bind_s(LDAP *ld,const PSTR  dn,const PCHAR cred,ULONG method);
+DECLSPEC_IMPORT ULONG WINAPI WLDAP32$ldap_search_s(LDAP *ld,PSTR base,ULONG scope,PSTR filter,PZPSTR attrs,ULONG attrsonly,PLDAPMessage *res);
+DECLSPEC_IMPORT ULONG WINAPI WLDAP32$ldap_count_entries(LDAP*,LDAPMessage*);
+
+DECLSPEC_IMPORT LDAPMessage*  WINAPI WLDAP32$ldap_first_entry(LDAP *ld,LDAPMessage *res);
+DECLSPEC_IMPORT LDAPMessage*  WINAPI WLDAP32$ldap_next_entry(LDAP*,LDAPMessage*);
+DECLSPEC_IMPORT PCHAR WINAPI WLDAP32$ldap_first_attribute(LDAP *ld,LDAPMessage *entry,BerElement **ptr);
+DECLSPEC_IMPORT ULONG WINAPI WLDAP32$ldap_count_values(PCHAR);
+DECLSPEC_IMPORT PCHAR WINAPI WLDAP32$ldap_get_values(LDAP *ld,LDAPMessage *entry,const PSTR attr);
+DECLSPEC_IMPORT ULONG WINAPI WLDAP32$ldap_value_free(PCHAR);
+DECLSPEC_IMPORT PCHAR WINAPI WLDAP32$ldap_next_attribute(LDAP *ld,LDAPMessage *entry,BerElement **ptr);
+DECLSPEC_IMPORT VOID WINAPI WLDAP32$ber_free(BerElement *pBerElement,INT fbuf);
+DECLSPEC_IMPORT VOID WINAPI WLDAP32$ldap_memfree(PCHAR);
+
+DECLSPEC_IMPORT ULONG WINAPI WLDAP32$ldap_unbind(LDAP*);
+DECLSPEC_IMPORT ULONG WINAPI WLDAP32$ldap_unbind_s(LDAP*);
+DECLSPEC_IMPORT ULONG WINAPI WLDAP32$ldap_msgfree(LDAPMessage*);
+
+DECLSPEC_IMPORT int __cdecl MSVCRT$strcmp(const char *_Str1,const char *_Str2);
+DECLSPEC_IMPORT PCHAR WINAPI MSVCRT$strstr(const char *haystack, const char *needle);
+
+#define TICKSTO1970         0x019db1ded53e8000LL
+#define TICKSTO1980         0x01a8e79fe1d58000LL
+#define TICKSPERSEC         10000000
+#ifdef _WIN64 //can't use builtin function for extended divition on x86
+DWORD GetTimeInSeconds(VOID)
+{
+    LARGE_INTEGER Time;
+    FILETIME FileTime;
+    DWORD dwSeconds;
+    int junk = 0;
+    lldiv_t test;
+    KERNEL32$GetSystemTimeAsFileTime(&FileTime);
+    Time.u.LowPart = FileTime.dwLowDateTime;
+    Time.u.HighPart = FileTime.dwHighDateTime;
+    Time.QuadPart = Time.QuadPart - TICKSTO1970;
+    Time.QuadPart = (uint64_t)Time.QuadPart / (uint64_t)TICKSPERSEC;
+    dwSeconds = Time.u.LowPart;
+
+    return dwSeconds;
+}
+#endif
+VOID ReturnDateTime(DWORD dwSeconds)
+{
+    LARGE_INTEGER Time;
+    FILETIME FileTime;
+    SYSTEMTIME SystemTime;
+    PCHAR DateBuffer[80];
+    PCHAR TimeBuffer[80];
+
+    Time.QuadPart = ((LONGLONG)dwSeconds * TICKSPERSEC) + TICKSTO1970;
+    
+    FileTime.dwLowDateTime = Time.u.LowPart;
+    FileTime.dwHighDateTime = Time.u.HighPart;
+    KERNEL32$FileTimeToLocalFileTime(&FileTime, &FileTime);
+    KERNEL32$FileTimeToSystemTime(&FileTime, &SystemTime);
+
+    KERNEL32$GetDateFormatW(LOCALE_USER_DEFAULT,
```

Comment:
For these two GetDatFormatW calls, please either switch your arrays to be wide character arrays(wchar_t) OR use the GetDateFormatA version.